submission.py

```diff
@@ -21,17 +23,43 @@ def _catalogued_action(state, player):
 
 
 def _planned_action(game, state, player):
-    return IterativeDeepening(
-        negamax,
-        timeout=game.timeout * TIMEOUT_BUFFER,
-        max_depth=game.rows * game.columns - state.counter,
-    )(
-        game=game,
-        state=state,
-        player=player,
-        heuristic_func=heuristic,
-        order_actions_func=order_actions,
-    )
+    result = None
+    timeout = game.timeout * TIMEOUT_BUFFER
+    min_depth = 0
+    max_depth = game.rows * game.columns - state.counter
+    _LOGGER.debug(f"Setting internal timeout: {timeout}.")
+
+    async def iterative_deepening():
```

Comment:
is there any reason you moved the iterative deepening logic into the `submission.py`?

Reply:
Mostly for simplicity, because the iterative deepening function gets to save intermediate results to the return value of _planned_action. It wouldn't be a problem to put the nested functions in a new function in iterative_deepening.py instead.

Reply:
* Inside the iterative deepening decorator